File: leetcode/027.remove-element.py
'''
leetcode #27 remove-element 移除元素
https://leetcode-cn.com/problems/remove-element/description/
'''
class Solution:
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        pos = 0
        for i in range(len(nums)):
            if nums[i] == val:
                continue
            nums[pos] = nums[i]
            pos += 1
        return pos

File: leetcode/027.remove-element.rs
impl Solution {
    pub fn remove_element(nums: &mut Vec<i32>, val: i32) -> i32 {
        let mut p1 = 0;
        let mut p2 = 0;
        while p2 < nums.len() {
            if nums[p2] != val {
                nums[p1] = nums[p2];
                p1 += 1;
            }
            p2 += 1;
        }
        return p1 as i32;
    }
}
